sdk/storage/storage-blob/src/StorageClient.ts

```diff
@@ -89,10 +89,11 @@ export abstract class StorageClient {
     for (const factory of this.pipeline.factories) {
       if (
         (isNode && factory instanceof StorageSharedKeyCredential) ||
-        factory instanceof AnonymousCredential ||
-        isTokenCredential(factory)
+        factory instanceof AnonymousCredential
       ) {
         this.credential = factory;
+      } else if (isTokenCredential((factory as any).credential)) {
```

Comment:
Could we introduce a class that extends `RequestPolicyFactory` then we could use `instanceof` check here?

Reply:
The factory is returned by core-http anoymously via `bearerTokenAuthenticationPolicy(credential, StorageOAuthScopes)`. Will you consider doing that in core-http?

That will definitely help simplify my checking in queue as well. https://github.com/Azure/azure-sdk-for-js/blob/ccffaafaa067eb15135269c6c9288b1c7ae1b7fa/sdk/storage/storage-queue/src/StorageClient.ts#L93

Reply:
Ah, right, we didn't want to expose the policy itself because of the weird factory model.

Reply:
would it be possible to add some wrapper in Storage? something like

```typescript
export function bearTokenAuthPolicyWrapper(credential: ..., scopes: ...): RequestPolicyFactoryWithCredential {
  const factory = bearerTokenAuthenticationPolicy(credential, scopes);
  return Object.defineProperty(factory, "credential", { value: credential });
}
```

Reply:
This may make it look nicer. But still doesn't help with the runtime type checking, right?
I need to merge this PR for now. We can continue the discussion on the improvement and make code changes if needed.